test/Microsoft.SqlTools.ServiceLayer.Test.Common/MemoryFileSystem.cs

```diff
@@ -12,17 +13,17 @@ public class MemoryFileSystem
 
         public static IFileStreamFactory GetFileStreamFactory()
         {
-            return GetFileStreamFactory(new Dictionary<string, byte[]>());
+            return GetFileStreamFactory(new ConcurrentDictionary<string, byte[]>());
         }
 
-        public static IFileStreamFactory GetFileStreamFactory(Dictionary<string, byte[]> storage)
+        public static IFileStreamFactory GetFileStreamFactory(ConcurrentDictionary<string, byte[]> storage)
         {
             Mock<IFileStreamFactory> mock = new Mock<IFileStreamFactory>();
             mock.Setup(fsf => fsf.CreateFile())
                 .Returns(() =>
                 {
                     string fileName = Guid.NewGuid().ToString();
-                    storage.Add(fileName, new byte[8192]);
+                    storage.TryAdd(fileName, new byte[8192]);
```

Comment:
>TryAdd [](start = 28, length = 6)

do we need to do anything if this fails?

Reply:
This only fails if the key already exists in the dictionary. Since we're creating a guid as the key in the line before, that will never happen. The only reason for the change is because `ConcurrentDictionary` doesn't expose a regular Add method